### ISSUE TITLE
Make platform-specific test aspects

### ISSUE THREAD

**Issue by jdegoes** [zio/zio#1541]
ZIO supports JVM and JS, and soon (hopefully) native. Some tests should only be run on some platforms, and some tests should have settings that depend on the platform.

We can provide test _aspects_ (which are _functions_ that _accept a spec_ and _return a spec_) to help people deal with these problems.

 - [ ] `TestAspect.jsOnly` — Used like `test("js-only test") { } @@ js`. Only runs the test on Javascript.
 - [ ] `TestAspect.jvmOnly` — Used like `test("js-only test") { } @@ jvm`. Only runs the test on JVM.
 - [ ] `TestAspect.js(aspect)` — Applies the specified aspect on Javascript. For example, `test(...) { } @@ js(timeout(60.seconds))` applies the `timeout(60.seconds)` combinator to the test, but only on Javascript.
 - [ ] `TestAspect.jvm(aspect)` — Applies the specified aspect on JVM. For example, `test(...) { } @@ jvm(nonFlaky(60.seconds))` applies the `nonFlaky(60.seconds)` combinator to the test, but only on the JVM.

Righting the logic of these should be straightforward; the boilerplate can be done by copy/paste of another simple test aspect.

The missing bit of functionality we need is something like `TestPlatform.isJVM` / `TestPlatform.isJS` helpers to tell running code if it is on the JVM or JS. These can be added by defining new `object` of the same name in `zio.test` package, one inside `js` root of `test`, and one inside `jvm` root of `test`. 

Then atop this these simple helpers, the above operators / combinators can be defined.

Finally, one more function inside `zio.test` package object could be useful:

```scala
def platformSpecific[R, E, A](js: => A, jvm: => A)(f: A => ZTest[R, E]): ZTest[R, E] = ???
```

It will pass the platform-specific information (e.g. `size` parameter) to the function, which will return a test that is returned from the function. If the platform is neither, then it returns a default ignored test.

**Comment by adamgfraser** [comment]
I'm working on it now.

**Comment by jdegoes** [comment]
@afamgfraser Awesome! ❤️🙏 Do you ever sleep? 😆

**Comment by mschuwalow** [comment]
Would it make sense to have the platform tests be baked in zio.system.System? We already have functions for reading properties there

**Comment by adamgfraser** [comment]
I think we might want to have it be part of `Platform` or `Runtime`. If we are needing to do platform specific configuration for ZIO Test our users will need to in some cases as well.

**Comment by ghostdogpr** [comment]
Closed by #1539
